codebuild/bin/test_exec_leak.sh

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env bash
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#  http://aws.amazon.com/apache2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+#
+
+set -e
+
+source codebuild/bin/s2n_setup_env.sh
+source codebuild/bin/jobs.sh
+
+function build() {
+  echo "=== BUILDING $1 ==="
+  cmake . -B$1 -DCMAKE_PREFIX_PATH=$TARGET_LIBCRYPTO_PATH ${@:2}
+  cmake --build $1 -- -j $JOBS
+}
+
+function fail() {
+    echo "test failure: $1"
+    exit 1
+}
+
+function write_exec_app() {
+cat <<EOF > build/detect_exec_leak.c
+#include <s2n.h>
+#include "unistd.h"
+
+int main() {
+    s2n_init();
+    execl("build/bin/detect_exec_leak_finish", "", NULL);
+    return 0;
+}
+EOF
+}
+
+function write_exec_finish_app() {
+cat <<EOF > build/detect_exec_leak_finish.c
+#include <s2n.h>
+
+int main() {
+    s2n_init();
+    s2n_cleanup();
+
+    /* close std* file descriptors so valgrind output is less noisy */
+    fclose(stdin);
+    fclose(stdout);
+    fclose(stderr);
+    return 0;
+}
+EOF
+}
+
+# download libcrypto if its not available
+WHICH_LIBCRYPTO=$(echo "${S2N_LIBCRYPTO:-"openssl-1.1.1"}")
```

Comment:
Do we expect this test to be running with multiple libcryptos? Can we just use the default one from the system path? If the flexibility isn't needed the codebuild stuff can be [way simpler](https://github.com/aws/s2n-tls/blob/main/codebuild/spec/buildspec_32bit_cross_compile.yml)

Reply:
Also are we specifically running this with openssl 1.1.1? What makes openssl 1.1.1 special?

Reply:
nit/opinion: would it make more sense to use AWS-LC? I think that's our "happy path" libcrypto. And then also add a comment that the choice of libcrypto is essentially arbitrary, and the test should work for any libcrypto.

Reply:
Since the libcrypto choice is somewhat arbitrary, it might make sense to just use whatever libcrypto is set in CI (S2N_LIBCRYPTO). This defaults to openssl if it's not set in codebuild:
https://github.com/aws/s2n-tls/blob/f9a6ffa8896c2a670f9e505a9448e1d29be8f989/codebuild/bin/s2n_setup_env.sh#L21

Reply:
Yep seeing that its actually defined a layer above. Going to simplify this and take whats set in the env.